torch/_decomp/decompositions.py

```diff
@@ -3754,6 +3755,53 @@ def _unsafe_masked_index_put_accumulate(x, mask, indices, values):
     return aten._unsafe_index_put(x, indices, masked_value, accumulate=True)
 
 
+@register_decomposition(aten.constant_pad_nd)
+@out_wrapper()
+def constant_pad_nd(
+    input: Tensor,
+    pad: Tuple[int, ...],
+    value: NumberType = 0,
+) -> Tensor:
+    torch._check(
+        len(pad) % 2 == 0,
+        lambda: "constant_pad_nd requires an even number of padding",
+    )
+    dim = len(pad) // 2
+    inp_shape = input.shape[-dim:]
+    nc_dim = input.dim() - dim
+
+    pad_left = [pad[2 * (dim - 1 - i)] for i in range(dim)]
+    pad_right = [pad[2 * (dim - 1 - i) + 1] for i in range(dim)]
+
+    out_indices = [
+        torch.arange(
+            -pad_left[i], inp_shape[i] + pad_right[i], device=input.device
+        ).reshape(-1, *[1] * (dim - 1 - i))
+        for i in range(dim)
+    ]
+
+    indices: List[Any] = [None] * input.dim()
+    for i in range(dim):
+        indices[i + nc_dim] = out_indices[i]
+
+    if builtins.all(pad <= 0 for pad in pad):
```

Comment:
For optimizations like this, you should use `torch.fx.experimental.symbolic_shapes.statically_known_true` which avoids introducing a guard.

Reply:
also, but [`for pad in pad`](https://en.wikipedia.org/wiki/Russell%27s_paradox) :P

Reply:
Is `statically_known_true` okay to have in `torch._decomp`?

Reply:
Yes, it's fine to use inside decompositions.